lib/workload/stateful/token_service/deploy/token-service-stack.ts

```diff
@@ -0,0 +1,181 @@
+      // 'cognito-idp:AdminEnableUser',
+      'cognito-idp:AdminGetUser',
+      'cognito-idp:AdminSetUserPassword',
+      'cognito-idp:InitiateAuth',
+      'cognito-idp:ListUserPools',
+      'cognito-idp:ListUsers',
+      'cognito-idp:SignUp'
+    );
+
+    // create token service user secret
+    const serviceUserSecret: Secret = new Secret(this, 'ServiceUserSecret', {
+      // just create a blank secret container first
+      // the admin should populate after enrolling the service user registration elsewhere
+      secretName: this.props.serviceUserSecretName,
+    });
+
+    serviceUserSecret.addRotationSchedule('ServiceUserRotationSchedule', {
+      rotationLambda: userRotationFn,
+      automaticallyAfter: Duration.days(7), // rotate service user cred every 7 days
+      rotateImmediatelyOnUpdate: true,
+    });
+
+    this.serviceUserSecret = serviceUserSecret;
+  }
+
+  private createJWTSecretWithRotation() {
+    const lambdaRole = new Role(this, 'JWTRole', {
+      assumedBy: new ServicePrincipal('lambda.amazonaws.com'),
+      description: 'Lambda execution role for ' + 'JWTRole',
+    });
+
+    const lambdaLogGroup = new LogGroup(this, 'JWTLogGroup');
+
+    // create service user cred rotation function
+    const jwtRotationFn = new PythonFunction(this, 'JWTRotateFn', {
+      entry: path.join(__dirname, '../token_service/'),
+      index: 'rotate_service_jwt.py',
+      handler: 'lambda_handler',
+      runtime: this.lambdaRuntimePythonVersion,
+      environment: {
+        ...this.lambdaEnv,
+        SERVICE_USER_SECRET_ID: this.props.serviceUserSecretName,
+        SERVICE_INFO_ENDPOINT: '', // TODO we need stable service info endpoint
+      },
+      vpc: this.vpc,
+      vpcSubnets: { subnets: this.vpc.privateSubnets },
+      architecture: Architecture.ARM_64,
+      timeout: Duration.seconds(28),
+      role: lambdaRole,
+      logGroup: lambdaLogGroup,
+    });
+    lambdaLogGroup.grantWrite(jwtRotationFn);
+
+    this.userPool.grant(jwtRotationFn, 'cognito-idp:InitiateAuth');
+
+    // create token service user jwt secret
+    const jwtSecret: Secret = new Secret(this, 'JWTSecret', {
+      // it should get populated by rotation
+      secretName: this.props.jwtSecretName,
+    });
+
+    jwtSecret.addRotationSchedule('ServiceUserRotationSchedule', {
+      rotationLambda: jwtRotationFn,
+      automaticallyAfter: Duration.hours(12), // rotate JWT every 12 hours
+      rotateImmediatelyOnUpdate: true,
+    });
+
+    this.serviceUserSecret.grantRead(jwtRotationFn.role as Role); // allow read to service user secret
+  }
+
+  private getUserPoolId(): IStringParameter {
+    /**
+     * FIXME One fine day in future when we have proper Cognito AAI setup.
+     *  For the moment, we leverage Portal and established Cognito infrastructure.
+     *  See https://github.com/umccr/orcabus/issues/102
+     */
+    const SSM_USER_POOL_ID: string = '/data_portal/client/cog_user_pool_id';
```

Comment:
Initially, feel, unsure about how we wanted to do with Cognito arrangement - whether to reuse the existing one or, spun up another Cognito instance afresh when we move on to UoM accounts... https://github.com/umccr/infrastructure/issues/402 

Anyhow, I will push them up, for now. It will end up SSM lookup either way down the track, I reckon.

Reply:
Alright, Cognito SSM param names have been pulled up to `constants.ts`.

Sharing this piece of information as constants should be ... fine. Feel, every other APIGW authoriser constructs will depend on these params. 

Again, sharing this is fine; as this part of system wide authN/Z param has to be harmonised across the board anyway.